packages/hardhat-core/src/internal/cli/vars.ts

```diff
@@ -0,0 +1,223 @@
+import chalk from "chalk";
+import debug from "debug";
+import { HardhatError } from "../core/errors";
+import { ERRORS } from "../core/errors-list";
+import { HardhatContext } from "../context";
+import { VarsManagerSetup } from "../core/vars/vars-manager-setup";
+import {
+  importCsjOrEsModule,
+  resolveConfigPath,
+} from "../core/config/config-loading";
+import { ArgumentsParser } from "./ArgumentsParser";
+
+const log = debug("hardhat:cli:vars");
+
+export async function handleVars(
+  allUnparsedCLAs: string[],
+  configPath: string | undefined
+): Promise<number> {
+  const { taskDefinition, taskArguments } =
+    await getTaskDefinitionAndTaskArguments(allUnparsedCLAs);
+
+  switch (taskDefinition.name) {
+    case "set":
+      return set(taskArguments.key, taskArguments.value);
+    case "get":
+      return get(taskArguments.key);
+    case "list":
+      return list();
+    case "delete":
+      return del(taskArguments.key);
+    case "path":
+      return path();
+    case "setup":
+      return setup(configPath);
+    default:
+      return 1; // Error code
+  }
+}
+
+async function set(key: string, value?: string): Promise<number> {
+  const varsManager = HardhatContext.getHardhatContext().varsManager;
+
+  varsManager.validateKey(key);
+
+  varsManager.set(key, value ?? (await getVarValue()));
+
+  console.warn(
```

Comment:
IMO this should be a log, not a warn.

Reply:
I used "warn" for all the additional info to avoid polluting the stdout. So if these commands are used via script only the useful values (e.g.: the value associated to a key) will be shown